### PR TITLE
Handle https://github.com/neo-project/neo/pull/3520

### PR DESCRIPTION
**Pull request by Hecate2**
# Description

Handles `VwEAwkpKAfoHdwARwG8AnXcAbwAl9////xHAzwJwlAAAdwDBwG8AnXcAbwAl9////0U=`

## Type of change

- [x] Optimization (the change is only an optimization)
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

invoke script `VwEAwkpKAfoHdwARwG8AnXcAbwAl9////xHAzwJwlAAAdwDBwG8AnXcAbwAl9////0U=`
Should finish in 0.4s when https://github.com/neo-project/neo/pull/3520 is merged.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by roman-khimov** [comment]
We better just do #3517 solving this and many other potential problems.

**Comment by Hecate2** [comment]
> We better just do #3517 solving this and many other potential problems.

I agree. Yet we can still do more to avoid GC. I have changed my strategy in this PR. We can check every dropped compound item to avoid graph searches.

File: src/Neo.VM/ExecutionEngine.cs [hunk 1]
```diff
@@ -288,7 +288,7 @@ public T Pop<T>() where T : StackItem
         /// </summary>
         protected virtual void PostExecuteInstruction(Instruction instruction)
         {
-            if (ReferenceCounter.Count < Limits.MaxStackSize) return;
+            if (ReferenceCounter.Count <= Limits.MaxStackSize) return;
```
Comment (Jim8y) [inline src/Neo.VM/ExecutionEngine.cs:291]: please make this one a seperate pr.

Comment (Hecate2) [reply src/Neo.VM/ExecutionEngine.cs:291]: I am still considering it. If <= is not applied, this PR makes no sense.

Comment (Jim8y) [reply src/Neo.VM/ExecutionEngine.cs:291]: > I am still considering it. If <= is not applied, this PR makes no sense.

you can make <= merhged first, then do this pr. If you mixed together, will be slower to review.

File: src/Neo.VM/ExecutionEngine.cs [hunk 2]
```diff
@@ -289,7 +289,7 @@ public T Pop<T>() where T : StackItem
...
```
Comment (Jim8y) [inline src/Neo.VM/ExecutionEngine.cs:292]:
```suggestion
            if (ReferenceCounter.Count < Limits.MaxStackSize) return;
```

**Comment by Jim8y** [comment]
We are replacing existing one with neogo gc, thus this pr is no longer needed.

**Pull request closed without merging**